validity/src/proposer.rs

```diff
@@ -1198,4 +1191,134 @@ where
+where
+    N: Network<UnsignedTx = TypedTransaction, TxEnvelope = TxEnvelope>,
+    N::TransactionRequest: TransactionBuilder4844,
+{
+    match proposer_signer {
+        ProposerSigner::Web3Signer(signer_url, signer_address) => {
+            // Set the from address to the signer address.
+            transaction_request.set_from(signer_address);
+
+            // Fill the transaction request with all of the relevant gas and nonce information.
+            let provider = ProviderBuilder::new().network::<N>().connect_http(l1_rpc);
+            let filled_tx = provider.fill(transaction_request).await?;
+
+            // Sign the transaction request using the Web3Signer.
+            let web3_provider = ProviderBuilder::new().network::<N>().connect_http(signer_url);
+            let signer = Web3Signer::new(web3_provider.clone(), signer_address);
+
+            let tx = filled_tx.as_builder().unwrap().clone();
+
+            // NOTE: This is a hack because there is not a "data" field on the TransactionRequest.
+            // `eth_signTransaction` expects a "data" field with the calldata.
+            // TODO: Once alloy fixes this, we can remove this wrapper.
+            let wrapper = TransactionRequestWrapper::<N>::new(tx);
+
+            let raw: Bytes =
+                signer.provider().client().request("eth_signTransaction", (wrapper,)).await?;
+
+            let tx_envelope = N::TxEnvelope::decode_2718(&mut raw.as_ref()).unwrap();
+
+            Ok(provider.send_tx_envelope(tx_envelope).await?)
+        }
+        ProposerSigner::LocalSigner(private_key) => {
+            let provider = ProviderBuilder::new()
+                .network::<N>()
+                .wallet(EthereumWallet::new(private_key.clone()))
+                .connect_http(l1_rpc);
+
+            // Set the from address to the Ethereum wallet address.
+            transaction_request.set_from(private_key.address());
+
+            // Fill the transaction request with all of the relevant gas and nonce information.
+            let filled_tx = provider.fill(transaction_request).await?;
+
+            Ok(provider.send_tx_envelope(filled_tx.as_envelope().unwrap().clone()).await?)
+        }
+    }
+}
+
+/// A wrapper around `TransactionRequest` that adds a data field as bytes.
+///
+/// This is a hack because there is not a "data" field on the TransactionRequest.
+/// `eth_signTransaction` expects a "data" field with the calldata.
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct TransactionRequestWrapper<N: Network> {
+    /// The underlying transaction request
+    #[serde(flatten)]
+    pub tx: N::TransactionRequest,
+    /// The transaction data as bytes
+    pub data: Option<Bytes>,
+}
+
+impl<N: Network> TransactionRequestWrapper<N> {
+    /// Create a new wrapper around a transaction request
+    pub fn new(tx: N::TransactionRequest) -> Self {
+        let data = tx.input().cloned();
+        Self { tx, data }
+    }
+}
```

Comment:
Based on [@mattsee](https://github.com/alloy-rs/alloy/issues/2378), can't you just set the `data` on the `TransactionRequest` manually until the next version of Alloy is released?

You should modify the test below too.

Reply:
Updated comment that https://github.com/alloy-rs/alloy/pull/2393 doesn't fix for N::TransactionRequest trait.